app/core/worker_pool.py

```
@@ -0,0 +1,56 @@
+from typing import Dict
+from ..core.config import settings
+from ..db.session import get_db
+
+class WorkerPool:
+    def __init__(self):
+        self.queue = asyncio.Queue(maxsize=settings.MAX_QUEUE_SIZE)
+        self.processing_status: Dict[str, Dict] = {}
+        self.workers = []
+
+    async def db_worker(self, worker_id: int):
+        db_session = await anext(get_db())
```

Ruff

Undefined name `anext`. Consider specifying `requires-python = ">= 3.10"` or `tool.ruff.target-version = "py310"` in your `pyproject.toml` file.

(F821)